src/Components/Spotify/TokenExtractor.js

```diff
@@ -0,0 +1,17 @@
+
+
+const tokenExtractor = function () {
+  const hash = window.location.hash
+    .substring(1)
+    .split("&")
+    .reduce((init, item)=>{
+      if (item) {
+        var parts = item.split("=");
+        init[parts[0]] = decodeURIComponent(parts[1]);
+      }
+      return init;
+    }, {});
+  return hash;
```

Comment:
Using `URLSearchParams`, this could be reduced to:

```javascript
const tokenExtractor = function () {
  const paramString = window.location.hash.substring(1);
  const params = new URLSearchParams(paramString);

  return Object.fromEntries(params);
}
```

Alternatively, `params` could be returned, although returning a `URLSearchParams` would change its usage a bit.

Reply:
This looks indeed better!